### PR TITLE
Update build, get rid of legacy, fix CLI

### LINKED ISSUE #434: ModelloCli is not working in 2.3.0
I cannot manage to make ModelloCli work in 2.3.0. I am getting
`SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/usr/share/java/slf4j/nop.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/share/java/slf4j/simple.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.helpers.NOPLoggerFactory]
Exception in thread "main" org.codehaus.plexus.component.repository.exception.ComponentLookupException: java.util.NoSuchElementException
      role: org.codehaus.modello.Modello
  roleHint: 
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:268)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:256)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:250)
	at org.codehaus.modello.ModelloCli.main(ModelloCli.java:43)
Caused by: java.util.NoSuchElementException
	at org.eclipse.sisu.inject.LocatedBeans$Itr.next(LocatedBeans.java:141)
	at org.eclipse.sisu.inject.LocatedBeans$Itr.next(LocatedBeans.java:76)
	at org.eclipse.sisu.plexus.DefaultPlexusBeans$Itr.next(DefaultPlexusBeans.java:76)
	at org.eclipse.sisu.plexus.DefaultPlexusBeans$Itr.next(DefaultPlexusBeans.java:56)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:264)
	... 3 more
`
There is either a problem in the lookup or (most likely) I am doing something wrong myself. I would not mind if one could help me with debugging this issue.

### PR DESCRIPTION
**Pull request by cstamas**
Fixes #434

### CODE REVIEW

File: modello-core/pom.xml [hunk 1]
```diff
@@ -19,47 +19,78 @@
     <dependency>
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
-      <scope>provided</scope>
```
Comment (cstamas) [inline modello-core/pom.xml:22]: Maybe "runtime" better, to make sure nothing picks this up.

**Comment by cstamas** [comment]
Not yet done: need to review (and probably) fix the modello-maven-plugin deps, as they are off now

File: modello-core/pom.xml (hunk 1 above)
Comment (cstamas) [reply modello-core/pom.xml:22]: is used... my bad

**Comment by fridrich** [comment]
I would still prefer https://github.com/codehaus-plexus/modello/pull/435/commits/90352d3cd8de0382be73d5ce16b5f9d96469e39c because there are tools out there that instantiate the Modello class. And also lookup of ModeloCore.ROLE is done all around the place in tests, so this would be consistent with that.

**Comment by cstamas** [comment]
@fridrich ported, pls review

**Comment by fridrich** [comment]
Another thing. the slf4j-simple does not need to be runtime dependency. one can have there slf4j-nop if one does not need logging

**Comment by cstamas** [comment]
Without slf4j backend, CLI will emit warning like you reported in the issue... So you say that no logging backend is needed at all? (noop?)

**Comment by fridrich** [comment]
The slf4j-nop is an artifact that is noop logger

**Comment by fridrich** [comment]
org.slf4j:slf4j-nop is the artifact

**Comment by fridrich** [comment]
So, one can leave the choice for the user.

**Comment by fridrich** [comment]
testing, will report in 30 minutes

**Comment by fridrich** [comment]
nagyom szepem keszenem

**Comment by cstamas** [comment]
ni za kaj (molim, i drugi put)

**Comment by fridrich** [comment]
> ni za kaj (molim, i drugi put)

In my original language it would be: "nemas za co (dakujem a aj druhy raz)"

**Comment by fridrich** [comment]
Just there is a downgrade of plexus-utils 4.0.1 -> 3.5.1 Intended?

**Comment by cstamas** [comment]
> Just there is a downgrade of plexus-utils 4.0.1 -> 3.5.1 Intended?

Yes, keep this Maven3 compat for now (p-u 3.x is for Maven3, p-u 4.x is for Maven4)

**Comment by cstamas** [comment]
Let's see what @slawekjaranowski has to say as well...

**Comment by cstamas** [comment]
@slawekjaranowski ping

**Review by slawekjaranowski — APPROVED** [review]
some magics with dependencies but ok 😄

**Comment by cstamas** [comment]
Reason for magic: we make modello-core "complete" as it provides now (and was providing) the CLI that needs to be runnable. And then later, in modello-maven-plugin we make sure that bits provided to maven plugins are properly provided/excluded. Before, it was "provided" in modello-core, so nothing was needed in modello-maven-plugin, but this breaks CLI. Now, modello-core is "complete" and modello-maven-plugin POM makes sure all is okay in plugin as well.

**Comment by fridrich** [comment]
Would a working CLI justify a 2.4.1? @slawekjaranowski @cstamas

**Pull request merged**